src/crate/client/connection.py

```diff
@@ -197,14 +198,21 @@ def get_blob_container(self, container_name):
 
     def _lowest_server_version(self):
         lowest = None
+        server_count = len(self.client.active_servers)
+        connection_errors = []
         for server in self.client.active_servers:
             try:
                 _, _, version = self.client.server_infos(server)
                 version = Version(version)
-            except (ValueError, ConnectionError):
+            except ConnectionError as ex:
+                connection_errors.append(ex)
+                continue
+            except (ValueError, InvalidVersion):
                 continue
             if not lowest or version < lowest:
                 lowest = version
+        if lowest is None and len(connection_errors) == server_count:
+            raise ConnectionError(str(connection_errors))
```

Comment:
`str(connection_errors)` uses an opaque way, just serializing the list of exception instances into a string, the Python way. Do you have any objections or suggestions to do it differently?

Reply:
> [...] suggestions to do it differently, like using JSON instead?

The procedure now uses JSON per e818221ac.